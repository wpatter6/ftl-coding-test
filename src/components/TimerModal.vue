<template>
  <modal name="timer-modal" :width="600" adaptive :click-to-close="false">
    <div class="timer__modal-body">
      <h4>Please input the number of time in seconds between emitting numbers and their frequency</h4>
      <input numeric-only ref="input" type="tel" v-model="selectedValue" />
      <button :class="parseInt(selectedValue) > 0 ? 'ok' : ''" @click="selectValue">Continue</button>
    </div>
  </modal>
</template>

<script lang="ts">
import { Component, Prop, Watch, Vue } from 'vue-property-decorator'

@Component
export default class TimerModal extends Vue {
  selectedValue: string

  @Prop() value: number

  constructor() {
    super()
    this.selectedValue = ''
  }

  show() {
    this.$modal.show('timer-modal')
    this.$emit('show')
    setTimeout(() => {
      const input = this.$refs.input as HTMLInputElement
      input.focus()
      input.select()
    }, 100)
  }

  selectValue() {
    if (parseInt(selectedValue) > 0) {
      this.$modal.hide('timer-modal')
      this.$emit('hide')
    }
  }

  mounted() {
    this.selectedValue = String(this.value)
  }

  @Watch('selelectedValue')
  selectedValueChanged(val: string) {
    this.$emit('input', parseInt(val, 10))
  }
}
</script>

<style scoped lang="scss">
.timer__modal-body {
  padding: 20px;
  display: flex;
  flex-direction: column;
}
</style>