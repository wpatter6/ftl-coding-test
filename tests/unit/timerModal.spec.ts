import { shallowMount } from '@vue/test-utils'
import TimerModal from '@/components/TimerModal.vue'
import flushPromises from 'flush-promises'
import Vue from 'vue'
import VModal from 'vue-js-modal'

Vue.use(VModal)

describe('TimerModal.vue', () => {
  it('renders props.value when passed', async () => {
    const value = '12'
    const wrapper = shallowMount(TimerModal, {
      propsData: { value },
    })

    const input = wrapper.find('input').element as HTMLInputElement

    await flushPromises()

    expect(input.value).toMatch(value)

    expect(wrapper).toMatchSnapshot()
  })
})
