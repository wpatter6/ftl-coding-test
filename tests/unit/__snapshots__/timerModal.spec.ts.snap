// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerModal.vue renders props.value when passed 1`] = `
<modal-stub name="timer-modal" delay="0" adaptive="true" overlaytransition="overlay-fade" classes="v--modal" minwidth="0" minheight="0" maxwidth="Infinity" maxheight="Infinity" width="600" height="300" pivotx="0.5" pivoty="0.5">
  <div class="timer__modal-body">
    <h4>Please input the number of time in seconds between emitting numbers and their frequency</h4> <input numeric-only="" type="tel"> <button class="ok">Continue</button>
  </div>
</modal-stub>
`;
